velox/dwio/common/SelectiveColumnReader.cpp

```diff
@@ -295,11 +295,11 @@ void SelectiveColumnReader::compactScalarValues<bool, bool>(
 char* SelectiveColumnReader::copyStringValue(folly::StringPiece value) {
   uint64_t size = value.size();
   if (stringBuffers_.empty() || rawStringUsed_ + size > rawStringSize_) {
-    if (!stringBuffers_.empty()) {
-      stringBuffers_.back()->setSize(rawStringUsed_);
-    }
     auto bytes = std::max(size, kStringBufferSize);
     BufferPtr buffer = AlignedBuffer::allocate<char>(bytes, &memoryPool_);
+    // Use the prefered size instead of the requested one to improve memory
```

Comment:
How is this improving efficiency? It seems a bit hacky to set size to be equal to capacity.

Reply:
The requested size is a little more than 16K and memory pool will give 24K for the buffer.  We want to use the full capacity here.

Tracking the real used size is possible but that will split the `setSize` call in multiple places.  I feel the added complexity (likely to create more bugs in future) would likely outweigh the benefit.